src/Microsoft.AspNetCore.SystemWebAdapters/Authentication/RemoteAppAuthenticationService.cs

```diff
@@ -23,19 +23,22 @@ internal partial class RemoteAppAuthenticationService : IRemoteAppAuthentication
     private readonly HttpClient _client;
     private readonly IAuthenticationResultFactory _resultFactory;
     private readonly ILogger<RemoteAppAuthenticationService> _logger;
-    private readonly IOptionsSnapshot<RemoteAppAuthenticationOptions> _optionsSnapshot;
+    private readonly IOptionsSnapshot<RemoteAppAuthenticationOptions> _authOptionsSnapshot;
+    private readonly RemoteAppOptions _remoteAppOptions;
     private RemoteAppAuthenticationOptions? _options;
 
     public RemoteAppAuthenticationService(
         HttpClient client,
         IAuthenticationResultFactory resultFactory,
-        IOptionsSnapshot<RemoteAppAuthenticationOptions> options,
+        IOptionsSnapshot<RemoteAppAuthenticationOptions> authOptions,
+        IOptions<RemoteAppOptions> remoteAppOptions,
```

Comment:
should we be consistent here in storing the optinos itself vs the IOptions<> container? Also, why is one thing using IOptionsSnapshop<> and another IOptions<>?

Reply:
Auth options are a special case because they're registered by name (each auth scheme will have its own options) and we don't know which scheme to get options for until `InitializeAsync` is called. We don't currently have multiple `RemoteAppOptions` registered in DI, so I just get those options in the ctor the usual way for things we don't expect to change. That pattern doesn't work for auth options, though, since we need to get them by name.